### PR TITLE
XML CodeLens preference page

### LINKED ISSUE #636: XML CodeLens preference page
Add a XML Codelens preference page to benefit with codelens inside XSD and DTD. See https://github.com/redhat-developer/vscode-xml/blob/master/docs/CodeLens.md

### PR DESCRIPTION
**Pull request by angelozerr**
XML CodeLens preference page

Fixes #636

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
With this PR you can enable XMl CodeLens inside XSD and DTD:

image

The click on code lens doesn't work because it requires to implement the proper command to execute the references. But we could support that in an another PR?

**Comment by mickaelistria** [comment]
I think it would be better to have an "Edition" preference page with the codelens entry.

File: org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/messages.properties [hunk 1]
```diff
@@ -21,6 +21,9 @@ XMLCatalogPreferencePage_OpenInEditorMessage=Do you want to Apply the changes an
 XMLCatalogPreferencePage_OpenInEditorApplyAndEdit=Apply and Edit
 XMLCatalogPreferencePage_OpenInEditorNo=No
 
+# XML CodeLens preference page
+XMLCodelensPreferencePage_codelens_enabled=Enable/disable XML &CodeLens
```
Comment (mickaelistria) [inline org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/messages.properties:25]: Just "Enable XML Codelens" as it's a checkbox, ideally also with a description of what are the expected codelens (reference count on schemas)

**Comment by angelozerr** [comment]
> I think it would be better to have an "Edition" preference page with the codelens entry.

In the future I would like to provide codelens settings per codelens kind (ex : xml.codeLens.schema.enabled,  xml.codeLens.dtd.enabled, etc), so I think it's better to have a Codelens page for that (to be consistent with JDT we have a CodeMining page).

File: org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/messages.properties (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/messages.properties:25]: > Just "Enable XML Codelens" as it's a checkbox, ideally also with a description of what are the expected codelens (reference count on schemas)

LemMinx provides this feature but LemMinx extension could provide Codelens too (ex : Codelens to execute a goal for maven), so it's not only restricted for XSD and DTD.

Comment (mickaelistria) [reply org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/messages.properties:25]: OK. It's more the "Enable/Disable" part which is not good UI workding besides a checkbox.

**Comment by mickaelistria** [comment]
Merged as f93fd6810a32e22a7e5e1218fb5135668ae67bd6 . Thanks!

**Pull request closed without merging**